gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/DatasetInfo.java

```diff
@@ -0,0 +1,409 @@
+      this.description = datasetInfo.description;
+      this.etag = datasetInfo.etag;
+      this.friendlyName = datasetInfo.friendlyName;
+      this.id = datasetInfo.id;
+      this.lastModified = datasetInfo.lastModified;
+      this.location = datasetInfo.location;
+      this.selfLink = datasetInfo.selfLink;
+    }
+
+    /**
+     * Sets the dataset identity.
+     */
+    public Builder datasetId(DatasetId datasetId) {
+      this.datasetId = checkNotNull(datasetId);
+      return this;
+    }
+
+    /**
+     * Sets the dataset's access control configuration.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/access-control">Access Control</a>
+     */
+    public Builder acl(List<Acl> acl) {
+      this.acl = acl != null ? ImmutableList.copyOf(acl) : null;
+      return this;
+    }
+
+    Builder creationTime(Long creationTime) {
+      this.creationTime = creationTime;
+      return this;
+    }
+
+    /**
+     * Sets the default lifetime of all tables in the dataset, in milliseconds. The minimum value is
+     * 3600000 milliseconds (one hour). Once this property is set, all newly-created tables in the
+     * dataset will have an expirationTime property set to the creation time plus the value in this
+     * property, and changing the value will only affect new tables, not existing ones. When the
+     * expirationTime for a given table is reached, that table will be deleted automatically. If a
+     * table's expirationTime is modified or removed before the table expires, or if you provide an
+     * explicit expirationTime when creating a table, that value takes precedence over the default
+     * expiration time indicated by this property.
+     */
+    public Builder defaultTableLifetime(Long defaultTableLifetime) {
+      this.defaultTableLifetime =
+          firstNonNull(defaultTableLifetime, Data.<Long>nullOf(Long.class));
+      return this;
+    }
+
+    /**
+     * Sets a user-friendly description for the dataset.
+     */
+    public Builder description(String description) {
+      this.description = firstNonNull(description, Data.<String>nullOf(String.class));
+      return this;
+    }
+
+    Builder etag(String etag) {
+      this.etag = etag;
+      return this;
+    }
+
+    /**
+     * Sets a user-friendly name for the dataset.
+     */
+    public Builder friendlyName(String friendlyName) {
+      this.friendlyName = firstNonNull(friendlyName, Data.<String>nullOf(String.class));
+      return this;
+    }
+
+    Builder id(String id) {
+      this.id = id;
+      return this;
+    }
+
+    Builder lastModified(Long lastModified) {
+      this.lastModified = lastModified;
+      return this;
+    }
+
+    /**
+     * Sets the geographic location where the dataset should reside.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/docs/managing_jobs_datasets_projects#dataset-location">
+     *     Dataset Location</a>
+     */
+    public Builder location(String location) {
```

Comment:
Same here. should we say experimental?
Also, the default value is mentioned in the [field description](https://cloud.google.com/bigquery/docs/reference/v2/datasets#access). is it also mentioned in the "@see" link?